src/Compilers/CSharp/Portable/Binder/Binder.NamespaceOrTypeOrAliasSymbolWithAnnotations.cs

```diff
@@ -9,21 +9,23 @@ internal partial class Binder
     {
         internal struct NamespaceOrTypeOrAliasSymbolWithAnnotations
         {
-            private readonly object _symbolOrTypeSymbolWithAnnotations;
+            private readonly TypeSymbolWithAnnotations _type;
+            private readonly Symbol _symbol;
 
-            private NamespaceOrTypeOrAliasSymbolWithAnnotations(object symbolOrTypeSymbolWithAnnotations)
+            private NamespaceOrTypeOrAliasSymbolWithAnnotations(TypeSymbolWithAnnotations type, Symbol symbol)
             {
-                Debug.Assert(symbolOrTypeSymbolWithAnnotations != null);
-                Debug.Assert(!(symbolOrTypeSymbolWithAnnotations is TypeSymbol));
-                _symbolOrTypeSymbolWithAnnotations = symbolOrTypeSymbolWithAnnotations;
+                Debug.Assert(type == null != (symbol is null));
```

Comment:
>type == null [](start = 29, length = 12)

How does `type == null` work?

Reply:
I've tried both: `IsNull` and `operator==(?, ?)`. The latter seemed slightly better since the result was closer to the existing code. But I'm open to either approach.

---
In reply to: [206357806](https://github.com/dotnet/roslyn/pull/28943#discussion_r206357806) [](ancestors = 206357806)

Reply:
It's not blocking this PR (ie. up to you), but I do think `IsNull` or `IsDefault` would be easier to follow. Already Jared and I tripped on this.

---
In reply to: [206539416](https://github.com/dotnet/roslyn/pull/28943#discussion_r206539416) [](ancestors = 206539416,206357806)